server/routerlicious/packages/services/src/redisSessionManager.ts

```diff
@@ -100,14 +119,36 @@ export class RedisCollaborationSessionManager implements ICollaborationSessionMa
 	}
 
 	public async getAllSessions(): Promise<ICollaborationSession[]> {
-		const sessionJsons = await this.redisClientConnectionManager
+		// Use HSCAN to iterate over te key:value pairs of the hashmap
+		// in batches to get all sessions with minimal impact on Redis.
+		const sessionJsonScanStream = this.redisClientConnectionManager
 			.getRedisClient()
-			.hgetall(this.prefix);
-		const sessions: ICollaborationSession[] = [];
-		for (const [fieldKey, sessionJson] of Object.entries(sessionJsons)) {
-			sessions.push(this.getFullSession(fieldKey, JSON.parse(sessionJson)));
-		}
-		return sessions;
+			.hscanStream(this.prefix, { count: this.options.maxScanBatchSize });
+		return new Promise((resolve, reject) => {
+			const sessions: ICollaborationSession[] = [];
+			sessionJsonScanStream.on("data", (result: string[]) => {
+				if (!result) {
```

Comment:
Do we need this? Wont the hscanStream event end with an end event when it is done instead of returning null?

Reply:
I actually went and looked at the ioredis source code when making sure I understood how to use hscanStream properly. I'm not certain about the inner workings of `Readable` from `node:stream` which ScanStream extends, but I do see a `push(null)`

https://github.com/redis/ioredis/blob/main/lib/ScanStream.ts#L25

It felt safer to do this than not, although I can look deeper into it if we feel this check is an important waste of processing power.

Reply:
oh please ignore it..just wanted to check if we can get rid of a couple of lines..not worth the effort to spend more time on this..lets leave it as it is